packages/backend-core/src/objectStore/objectStore.ts

```diff
@@ -138,9 +154,9 @@ export async function createBucketIfNotExists(
           .promise()
         await promises[bucketName]
         delete promises[bucketName]
-        return true
+        return { created: true, exists: false }
       } else {
-        throw new Error("Access denied to object store bucket.")
+        throw new Error("Access denied to object store bucket." + err)
```

Comment:
I'd recommend using the `cause` param here instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#rethrowing_an_error_with_a_cause, it'll preserve the stack trace of the cause that way.

```js
> const e1 = new Error("foo")
undefined
> const e2 = new Error("bar", { cause: e1 })
undefined
> e1
Error: foo
    at REPL1:1:12
    at ContextifyScript.runInThisContext (node:vm:121:12)
    at REPLServer.defaultEval (node:repl:599:22)
    at bound (node:domain:432:15)
    at REPLServer.runBound [as eval] (node:domain:443:12)
    at REPLServer.onLine (node:repl:929:10)
    at REPLServer.emit (node:events:526:35)
    at REPLServer.emit (node:domain:488:12)
    at [_onLine] [as _onLine] (node:internal/readline/interface:416:12)
    at [_line] [as _line] (node:internal/readline/interface:887:18)
> e2
Error: bar
    at REPL2:1:12
    at ContextifyScript.runInThisContext (node:vm:121:12)
    ... 7 lines matching cause stack trace ...
    at [_line] [as _line] (node:internal/readline/interface:887:18) {
  [cause]: Error: foo
      at REPL1:1:12
      at ContextifyScript.runInThisContext (node:vm:121:12)
      at REPLServer.defaultEval (node:repl:599:22)
      at bound (node:domain:432:15)
      at REPLServer.runBound [as eval] (node:domain:443:12)
      at REPLServer.onLine (node:repl:929:10)
      at REPLServer.emit (node:events:526:35)
      at REPLServer.emit (node:domain:488:12)
      at [_onLine] [as _onLine] (node:internal/readline/interface:416:12)
      at [_line] [as _line] (node:internal/readline/interface:887:18)
}
```

Reply:
This is only available with TS 2022 declarations - we only are at 2020 having a look at `tsconfig.build.json`. And I don't feel comfortable just changing that for this one thing, might have larger ramifications and break stuff?